### ISSUE TITLE
up to three threads run the same job in apscheduler

### ISSUE THREAD

**Issue by Tanych** [gwu-libraries/sfm-ui#274]
For one job, sometimes there are two or three threads to run the same job. It might be something wrong with the  apscheduler. I am trying to update the apscheduler to 3.0.6 or 3.1.0 to confirm the issue.
```python
[Thu May 19 13:17:03.123651 2016] [:error] [pid 220:tid 139973221107456] INFO 2016-05-19 13:17:03,123 220 apscheduler.executors.default Job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:47:03 EST)" executed successfully
[Thu May 19 13:17:03.125289 2016] [:error] [pid 220:tid 139972845762304] INFO 2016-05-19 13:17:03,125 220 apscheduler.executors.default Job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:47:03 EST)" executed successfully
```

The harvester shows two or three same logs:

```python
Type	Date requested	Status	Messages
Weibo timeline	May 20, 2016, 9:47 a.m.	Success	0 messages
Weibo timeline	May 20, 2016, 9:47 a.m.	Success	0 messages
```

**Comment by justinlittman** [comment]
I wonder if the problem is that the apscheduler is running in the ui and the message consumer.  (Not sure what the 3rd is.)

**Comment by Tanych** [comment]
I have updated the apscheduler to 3.0.6 in my testing environment.  I will check it tomorrow.

**Comment by Tanych** [comment]
@justinlittman Currently, the five harvesters have no duplicate logs. I will figure it out on tomorrow with more data.

**Comment by justinlittman** [comment]
Great. Let me know if you need any assistance.  If think what we're going to want to do is make it so that the apscheduler only runs for sfm-ui and does not run when management commands are run.

**Comment by Tanych** [comment]
logs detail of sfmdevapp_1 for 3 times running

```python```
[Thu May 19 13:17:03.067047 2016] [:error] [pid 220:tid 139973221107456] INFO 2016-05-19 13:17:03,066 220 apscheduler.executors.default Running job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:17:03 EST)" (scheduled at 2016-05-19 13:17:03.058081-05:00)
sfmdevapp_1               | INFO 2016-05-19 13:17:03,073 74 apscheduler.executors.default Running job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:17:03 EST)" (scheduled at 2016-05-19 13:17:03.058081-05:00)
sfmdevapp_1               | [Thu May 19 13:17:03.076971 2016] [:error] [pid 220:tid 139972845762304] INFO 2016-05-19 13:17:03,076 220 apscheduler.executors.default Running job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:17:03 EST)" (scheduled at 2016-05-19 13:17:03.058081-05:00)
sfmdevapp_1               | DEBUG 2016-05-19 13:17:03,083 74 ui.jobs Sending weibo_timeline message to harvest.start.weibo.weibo_timeline with id d9b80ffcb6074af6a2ea6a9409c3b12e
sfmdevapp_1               | DEBUG 2016-05-19 13:17:03,083 74 ui.rabbit Sending message to harvest.start.weibo.weibo_timeline: {
sfmdevapp_1               |     "collection": {
sfmdevapp_1               |         "id": "6110b42abacd46bd947681d8b6a9aa22"
sfmdevapp_1               |     }, 
sfmdevapp_1               |     "id": "d9b80ffcb6074af6a2ea6a9409c3b12e", 
sfmdevapp_1               |     "credentials": {
sfmdevapp_1               |         "access_token": "2.00kQCxKGpYiFYDc41039481c0wvwei", 
sfmdevapp_1               |         "api_secret": "4c6f6c568ad9ef6b6a81c42b172707b1", 
sfmdevapp_1               |         "api_key": "3252156855", 
sfmdevapp_1               |         "redirect_uri": "https://www.google.com/"
sfmdevapp_1               |     }, 
sfmdevapp_1               |     "path": "/sfm-data/collection/6110b42abacd46bd947681d8b6a9aa22/027c4d4c0df240749fb99ea1e30abdab", 
sfmdevapp_1               |     "type": "weibo_timeline", 
sfmdevapp_1               |     "options": {
sfmdevapp_1               |         "incremental": true
sfmdevapp_1               |     }
sfmdevapp_1               | }
sfmdevapp_1               | [Thu May 19 13:17:03.095727 2016] [:error] [pid 220:tid 139973221107456] DEBUG 2016-05-19 13:17:03,095 220 ui.jobs Sending weibo_timeline message to harvest.start.weibo.weibo_timeline with id ec2ba6bdf3a34407b44aa40baba34bb4
sfmdevapp_1               | [Thu May 19 13:17:03.097479 2016] [:error] [pid 220:tid 139973221107456] DEBUG 2016-05-19 13:17:03,095 220 ui.rabbit Sending message to harvest.start.weibo.weibo_timeline: {
sfmdevapp_1               | [Thu May 19 13:17:03.097490 2016] [:error] [pid 220:tid 139973221107456]     "collection": {
sfmdevapp_1               | [Thu May 19 13:17:03.097493 2016] [:error] [pid 220:tid 139973221107456]         "id": "6110b42abacd46bd947681d8b6a9aa22"
sfmdevapp_1               | [Thu May 19 13:17:03.097495 2016] [:error] [pid 220:tid 139973221107456]     }, 
sfmdevapp_1               | [Thu May 19 13:17:03.097496 2016] [:error] [pid 220:tid 139973221107456]     "id": "ec2ba6bdf3a34407b44aa40baba34bb4", 
sfmdevapp_1               | [Thu May 19 13:17:03.097498 2016] [:error] [pid 220:tid 139973221107456]     "credentials": {
sfmdevapp_1               | [Thu May 19 13:17:03.097499 2016] [:error] [pid 220:tid 139973221107456]         "access_token": "2.00kQCxKGpYiFYDc41039481c0wvwei", 
sfmdevapp_1               | [Thu May 19 13:17:03.097501 2016] [:error] [pid 220:tid 139973221107456]         "api_secret": "4c6f6c568ad9ef6b6a81c42b172707b1", 
sfmdevapp_1               | [Thu May 19 13:17:03.097503 2016] [:error] [pid 220:tid 139973221107456]         "api_key": "3252156855", 
sfmdevapp_1               | [Thu May 19 13:17:03.097504 2016] [:error] [pid 220:tid 139973221107456]         "redirect_uri": "https://www.google.com/"
sfmdevapp_1               | [Thu May 19 13:17:03.097506 2016] [:error] [pid 220:tid 139973221107456]     }, 
sfmdevapp_1               | [Thu May 19 13:17:03.097507 2016] [:error] [pid 220:tid 139973221107456]     "path": "/sfm-data/collection/6110b42abacd46bd947681d8b6a9aa22/027c4d4c0df240749fb99ea1e30abdab", 
sfmdevapp_1               | [Thu May 19 13:17:03.097512 2016] [:error] [pid 220:tid 139973221107456]     "type": "weibo_timeline", 
sfmdevapp_1               | [Thu May 19 13:17:03.097514 2016] [:error] [pid 220:tid 139973221107456]     "options": {
sfmdevapp_1               | [Thu May 19 13:17:03.097516 2016] [:error] [pid 220:tid 139973221107456]         "incremental": true
sfmdevapp_1               | [Thu May 19 13:17:03.097517 2016] [:error] [pid 220:tid 139973221107456]     }
sfmdevapp_1               | [Thu May 19 13:17:03.097518 2016] [:error] [pid 220:tid 139973221107456] }
sfmdevapp_1               | [Thu May 19 13:17:03.105270 2016] [:error] [pid 220:tid 139972845762304] DEBUG 2016-05-19 13:17:03,105 220 ui.jobs Sending weibo_timeline message to harvest.start.weibo.weibo_timeline with id 59f2f1cd0bc345178b5f3fdbd77ddb4c
sfmdevapp_1               | [Thu May 19 13:17:03.105552 2016] [:error] [pid 220:tid 139972845762304] DEBUG 2016-05-19 13:17:03,105 220 ui.rabbit Sending message to harvest.start.weibo.weibo_timeline: {
sfmdevapp_1               | [Thu May 19 13:17:03.105562 2016] [:error] [pid 220:tid 139972845762304]     "collection": {
sfmdevapp_1               | [Thu May 19 13:17:03.105564 2016] [:error] [pid 220:tid 139972845762304]         "id": "6110b42abacd46bd947681d8b6a9aa22"
sfmdevapp_1               | [Thu May 19 13:17:03.105566 2016] [:error] [pid 220:tid 139972845762304]     }, 
sfmdevapp_1               | [Thu May 19 13:17:03.105568 2016] [:error] [pid 220:tid 139972845762304]     "id": "59f2f1cd0bc345178b5f3fdbd77ddb4c", 
sfmdevapp_1               | [Thu May 19 13:17:03.105569 2016] [:error] [pid 220:tid 139972845762304]     "credentials": {
sfmdevapp_1               | [Thu May 19 13:17:03.105571 2016] [:error] [pid 220:tid 139972845762304]         "access_token": "2.00kQCxKGpYiFYDc41039481c0wvwei", 
sfmdevapp_1               | [Thu May 19 13:17:03.105572 2016] [:error] [pid 220:tid 139972845762304]         "api_secret": "4c6f6c568ad9ef6b6a81c42b172707b1", 
sfmdevapp_1               | [Thu May 19 13:17:03.105574 2016] [:error] [pid 220:tid 139972845762304]         "api_key": "3252156855", 
sfmdevapp_1               | [Thu May 19 13:17:03.105575 2016] [:error] [pid 220:tid 139972845762304]         "redirect_uri": "https://www.google.com/"
sfmdevapp_1               | [Thu May 19 13:17:03.105581 2016] [:error] [pid 220:tid 139972845762304]     }, 
sfmdevapp_1               | [Thu May 19 13:17:03.105582 2016] [:error] [pid 220:tid 139972845762304]     "path": "/sfm-data/collection/6110b42abacd46bd947681d8b6a9aa22/027c4d4c0df240749fb99ea1e30abdab", 
sfmdevapp_1               | [Thu May 19 13:17:03.105584 2016] [:error] [pid 220:tid 139972845762304]     "type": "weibo_timeline", 
sfmdevapp_1               | [Thu May 19 13:17:03.105585 2016] [:error] [pid 220:tid 139972845762304]     "options": {
sfmdevapp_1               | [Thu May 19 13:17:03.105587 2016] [:error] [pid 220:tid 139972845762304]         "incremental": true
sfmdevapp_1               | [Thu May 19 13:17:03.105588 2016] [:error] [pid 220:tid 139972845762304]     }
sfmdevapp_1               | [Thu May 19 13:17:03.105590 2016] [:error] [pid 220:tid 139972845762304] }
sfmdevapp_1               | INFO 2016-05-19 13:17:03,112 74 apscheduler.executors.default Job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:47:03 EST)" executed successfully
sfmdevapp_1               | [Thu May 19 13:17:03.123651 2016] [:error] [pid 220:tid 139973221107456] INFO 2016-05-19 13:17:03,123 220 apscheduler.executors.default Job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:47:03 EST)" executed successfully
sfmdevapp_1               | [Thu May 19 13:17:03.125289 2016] [:error] [pid 220:tid 139972845762304] INFO 2016-05-19 13:17:03,125 220 apscheduler.executors.default Job "Harvest (30) for seedset 2 (trigger: interval[0:30:00], next run at: 2016-05-19 13:47:03 EST)" executed successfully
sfmdevapp_1               | DEBUG 2016-05-19 13:17:12,130 74 message_consumer.sfm_ui_consumer Warc with id 3d40e47846f2435c815778855205d2c0
sfmdevapp_1               | DEBUG 2016-05-19 13:17:13,137 74 message_consumer.sfm_ui_consumer Updating harvest with id d9b80ffcb6074af6a2ea6a9409c3b12e
sfmdevapp_1               | DEBUG 2016-05-19 13:17:13,150 74 ui.sched Skipping scheduling harvest of seedset 2 since nothing significant changed.
sfmdevapp_1               | DEBUG 2016-05-19 13:17:18,658 74 message_consumer.sfm_ui_consumer Warc with id 02aa0497ba66432992022b175b92cadc
sfmdevapp_1               | DEBUG 2016-05-19 13:17:19,665 74 message_consumer.sfm_ui_consumer Updating harvest with id ec2ba6bdf3a34407b44aa40baba34bb4
sfmdevapp_1               | DEBUG 2016-05-19 13:17:19,677 74 ui.sched Skipping scheduling harvest of seedset 2 since nothing significant changed.
sfmdevapp_1               | DEBUG 2016-05-19 13:17:25,186 74 message_consumer.sfm_ui_consumer Warc with id 497db20d2b384230ac0c749d74996680
sfmdevapp_1               | DEBUG 2016-05-19 13:17:26,192 74 message_consumer.sfm_ui_consumer Updating harvest with id 59f2f1cd0bc345178b5f3fdbd77ddb4c
sfmdevapp_1               | DEBUG 2016-05-19 13:17:26,204 74 ui.sched Skipping scheduling harvest of seedset 2 since nothing significant changed.

```

**Comment by Tanych** [comment]
@justinlittman  After updating the apscheduler to 3.0.6, the 100+ harvesters have no duplicate.  We can consider this temporary solution.

**Comment by kerchner** [comment]
As discussed: Suggest adding a configuration flag to run apscheduler (if not present, do not run).   Remove `apscheduler` from `INSTALLED_APPS`; add back if flag is set.  We think this will eliminate any apschedulers that may not have shut down from previous management commands in `invoke.sh`.

It is also possible that multiple `apscheduler` instances are being run due to the "funky" apache2 startup steps in `invoke.sh`.  Recommend revisiting these.

**Comment by kerchner** [comment]
Fixed by #284
